Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -139,17 +163,81 @@ BTDiagnostics::InitializeFieldBufferData ( int i_buffer , int lev)
 {
     auto & warpx = WarpX::GetInstance();
     // 1. Lab-frame time for the i^th snapshot
+    m_t_lab[i_buffer] = i_buffer * m_dt_snapshots_lab;
+    // 2. Define domain in boosted frame at level, lev
+    amrex::RealBox diag_dom;
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim ) {
+        diag_dom.setLo(idim, std::max(m_lo[idim],warpx.Geom(lev).ProbLo(idim)) );
+        diag_dom.setHi(idim, std::min(m_hi[idim],warpx.Geom(lev).ProbHi(idim)) );
+    }
     // 2. Define domain in boosted frame at level, lev
     // 3. Initializing the m_buffer_box for the i^th snapshot.
     //    At initialization, the Box has the same index space as the boosted-frame
     //    As time-progresses, the z-dimension indices will be modified based on
     //    current_z_lab
+    amrex::IntVect lo(0);
+    amrex::IntVect hi(1);
+    for (int idim=0; idim < AMREX_SPACEDIM; ++idim) {
+        // lo index with same cell-size as simulation at level, lev.
+        lo[idim] = std::max( static_cast<int>( floor (
+                      ( diag_dom.lo(idim) - warpx.Geom(lev).ProbLo(idim)) /
+                        warpx.Geom(lev).CellSize(idim)) ), 0 );
+        // hi index with same cell-size as simulation at level, lev.
+        hi[idim] = std::max( static_cast<int> ( ceil (
+                      ( diag_dom.hi(idim) - warpx.Geom(lev).ProbLo(idim)) /
+                        warpx.Geom(lev).CellSize(idim) ) ), 0) - 1 ;
```

Comment:
That was a bit hard to read for me :)
```suggestion
        lo[idim] = std::max(
            0,
            static_cast<int>( floor(
                ( diag_dom.lo(idim) - warpx.Geom(lev).ProbLo(idim) ) /
                  warpx.Geom(lev).CellSize(idim)
            ))
        );
        // hi index with same cell-size as simulation at level, lev.
        hi[idim] = std::max(
            0,
            static_cast<int>( ceil(
                ( diag_dom.hi(idim) - warpx.Geom(lev).ProbLo(idim) ) /
                  warpx.Geom(lev).CellSize(idim)
            ))
        ) - 1;
```

One could make the 2nd argument potentially a named, temporary, const variable :)